ICE: ` expect >1 RPITITs in call to 'collect_return_position_impl_trait_in_trait_tys'`

<!--
Thank you for finding an Internal Compiler Error! 🧊  If possible, try to provide
a minimal verifiable example. You can read "Rust Bug Minimization Patterns" for
how to create smaller examples.
http://blog.pnkfx.org/blog/2019/11/18/rust-bug-minimization-patterns/
-->

### Code
`rustc_with_debug_assertions -Zlower-impl-trait-in-trait-to-assoc-ty file.rs`
```Rust
#![feature(return_position_impl_trait_in_trait)]
#![feature(specialization)]

trait Foo {
    fn bar() -> Wrapper<impl Sized>;
}

impl Foo for () {
    fn bar() -> i32 {
        0
    }
}

fn main() {}

```


### Meta
<!--
If you're using the stable version of the compiler, you should also check if the
bug also exists in the beta or nightly versions.
-->

`rustc --version --verbose`:
```
0d50ab77397d7278500abd22d34de0e10940b2ee
```

### Error output

```
error[E0412]: cannot find type `Wrapper` in this scope
 --> /home/matthias/vcs/github/rust/tests/ui/impl-trait/icemaker/5BA695890944A2D1E5C0D962DDA4006BABC66F27CECCC0A6A1719F8BAFC9F84C.rs:5:17
  |
5 |     fn bar() -> Wrapper<impl Sized>;
  |                 ^^^^^^^ not found in this scope
```

<!--
Include a backtrace in the code block by setting `RUST_BACKTRACE=1` in your
environment. E.g. `RUST_BACKTRACE=1 cargo build`.
-->
<details><summary><strong>Backtrace</strong></summary>
<p>

```
thread 'rustc' panicked at 'assertion failed: `(left != right)`
  left: `0`,
 right: `0`: expect >1 RPITITs in call to `collect_return_position_impl_trait_in_trait_tys`', compiler/rustc_hir_analysis/src/check/compare_impl_item.rs:672:5
stack backtrace:
   0:     0x7f9a67a826e6 - std::backtrace_rs::backtrace::libunwind::trace::h5fea0fc64dda8363
                               at /home/matthias/vcs/github/rust_debug_assertions/library/std/src/../../backtrace/src/backtrace/libunwind.rs:93:5
   1:     0x7f9a67a826e6 - std::backtrace_rs::backtrace::trace_unsynchronized::h6f2c506bd5b9d9e9
                               at /home/matthias/vcs/github/rust_debug_assertions/library/std/src/../../backtrace/src/backtrace/mod.rs:66:5
   2:     0x7f9a67a826e6 - std::sys_common::backtrace::_print_fmt::hd228effaa0d25ec8
                               at /home/matthias/vcs/github/rust_debug_assertions/library/std/src/sys_common/backtrace.rs:65:5
   3:     0x7f9a67a826e6 - <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt::h427ae8254cf55f9a
                               at /home/matthias/vcs/github/rust_debug_assertions/library/std/src/sys_common/backtrace.rs:44:22
   4:     0x7f9a67b543c8 - core::fmt::rt::Argument::fmt::h4f8c28fcc5e0c6ce
                               at /home/matthias/vcs/github/rust_debug_assertions/library/core/src/fmt/rt.rs:138:9
   5:     0x7f9a67b543c8 - core::fmt::write::h7d46c21d618fb808
                               at /home/matthias/vcs/github/rust_debug_assertions/library/core/src/fmt/mod.rs:1094:21
   6:     0x7f9a67ab8eef - std::io::Write::write_fmt::hc07c60e60e0f5122
                               at /home/matthias/vcs/github/rust_debug_assertions/library/std/src/io/mod.rs:1714:15
   7:     0x7f9a67a824f5 - std::sys_common::backtrace::_print::h28dec6f87b410b80
                               at /home/matthias/vcs/github/rust_debug_assertions/library/std/src/sys_common/backtrace.rs:47:5
   8:     0x7f9a67a824f5 - std::sys_common::backtrace::print::hcb2cfe4ac1a83e6e
                               at /home/matthias/vcs/github/rust_debug_assertions/library/std/src/sys_common/backtrace.rs:34:9
   9:     0x7f9a67aecfcc - std::panicking::default_hook::{{closure}}::he97651026a7197e3
  10:     0x7f9a67aecc87 - std::panicking::default_hook::hf3b979bbeffd9a34
                               at /home/matthias/vcs/github/rust_debug_assertions/library/std/src/panicking.rs:288:9
  11:     0x7f9a6992529b - <alloc[59fef786b468a15e]::boxed::Box<dyn for<'a, 'b> core[ad2aefeb2646d5e0]::ops::function::Fn<(&'a core[ad2aefeb2646d5e0]::panic::panic_info::PanicInfo<'b>,), Output = ()> + core[ad2aefeb2646d5e0]::marker::Sync + core[ad2aefeb2646d5e0]::marker::Send> as core[ad2aefeb2646d5e0]::ops::function::Fn<(&core[ad2aefeb2646d5e0]::panic::panic_info::PanicInfo,)>>::call
                               at /home/matthias/vcs/github/rust_debug_assertions/library/alloc/src/boxed.rs:2024:9
  12:     0x7f9a6992529b - rustc_driver_impl[5def55cd179e60aa]::install_ice_hook::{closure#0}
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_driver_impl/src/lib.rs:1342:13
  13:     0x7f9a67aed7b5 - <alloc::boxed::Box<F,A> as core::ops::function::Fn<Args>>::call::h53bbeb1cc07f5b84
                               at /home/matthias/vcs/github/rust_debug_assertions/library/alloc/src/boxed.rs:2024:9
  14:     0x7f9a67aed7b5 - std::panicking::rust_panic_with_hook::hb6e9daab1c650917
                               at /home/matthias/vcs/github/rust_debug_assertions/library/std/src/panicking.rs:709:13
  15:     0x7f9a67a82bf7 - std::panicking::begin_panic_handler::{{closure}}::hb7adf0d88631ceba
                               at /home/matthias/vcs/github/rust_debug_assertions/library/std/src/panicking.rs:597:13
  16:     0x7f9a67a827c6 - std::sys_common::backtrace::__rust_end_short_backtrace::h7ec748af3e9a65cc
                               at /home/matthias/vcs/github/rust_debug_assertions/library/std/src/sys_common/backtrace.rs:151:18
  17:     0x7f9a67aed322 - rust_begin_unwind
                               at /home/matthias/vcs/github/rust_debug_assertions/library/std/src/panicking.rs:593:5
  18:     0x7f9a67a7a3e3 - core::panicking::panic_fmt::hb3d3573ea68cbca7
                               at /home/matthias/vcs/github/rust_debug_assertions/library/core/src/panicking.rs:67:14
  19:     0x7f9a67a7a95b - core::panicking::assert_failed_inner::h106e6f9e0ca38e10
  20:     0x7f9a68687609 - core[ad2aefeb2646d5e0]::panicking::assert_failed::<usize, usize>
                               at /home/matthias/vcs/github/rust_debug_assertions/library/core/src/panicking.rs:229:5
  21:     0x7f9a69df1775 - rustc_hir_analysis[38d60f9cb4d84cc8]::check::compare_impl_item::collect_return_position_impl_trait_in_trait_tys::{closure#0}
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_hir_analysis/src/check/compare_impl_item.rs:672:5
  22:     0x7f9a69df1775 - rustc_hir_analysis[38d60f9cb4d84cc8]::check::compare_impl_item::collect_return_position_impl_trait_in_trait_tys
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_hir_analysis/src/check/compare_impl_item.rs:610:1
  23:     0x7f9a6b42c665 - rustc_query_impl[9dd290307928cbc3]::query_impl::collect_return_position_impl_trait_in_trait_tys::dynamic_query::{closure#2}::{closure#0}
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_query_impl/src/plumbing.rs:582:47
  24:     0x7f9a6b42c665 - rustc_query_impl[9dd290307928cbc3]::plumbing::__rust_begin_short_backtrace::<rustc_query_impl[9dd290307928cbc3]::query_impl::collect_return_position_impl_trait_in_trait_tys::dynamic_query::{closure#2}::{closure#0}, rustc_middle[6b5778e8ec0a50b1]::query::erase::Erased<[u8; 8usize]>>
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_query_impl/src/plumbing.rs:506:18
  25:     0x7f9a6b5f4a3f - rustc_query_impl[9dd290307928cbc3]::query_impl::collect_return_position_impl_trait_in_trait_tys::dynamic_query::{closure#2}
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_query_impl/src/plumbing.rs:578:25
  26:     0x7f9a6b5f4a3f - <rustc_query_impl[9dd290307928cbc3]::query_impl::collect_return_position_impl_trait_in_trait_tys::dynamic_query::{closure#2} as core[ad2aefeb2646d5e0]::ops::function::FnOnce<(rustc_middle[6b5778e8ec0a50b1]::ty::context::TyCtxt, rustc_span[93bd9c9d5a358720]::def_id::DefId)>>::call_once
                               at /home/matthias/vcs/github/rust_debug_assertions/library/core/src/ops/function.rs:250:5
  27:     0x7f9a6b38b7cf - <rustc_query_impl[9dd290307928cbc3]::DynamicConfig<rustc_query_system[67bbb282021fa3e9]::query::caches::DefaultCache<rustc_span[93bd9c9d5a358720]::def_id::DefId, rustc_middle[6b5778e8ec0a50b1]::query::erase::Erased<[u8; 8usize]>>, false, false, false> as rustc_query_system[67bbb282021fa3e9]::query::config::QueryConfig<rustc_query_impl[9dd290307928cbc3]::plumbing::QueryCtxt>>::compute
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_query_impl/src/lib.rs:116:9
  28:     0x7f9a6b38b7cf - rustc_query_system[67bbb282021fa3e9]::query::plumbing::execute_job_non_incr::<rustc_query_impl[9dd290307928cbc3]::DynamicConfig<rustc_query_system[67bbb282021fa3e9]::query::caches::DefaultCache<rustc_span[93bd9c9d5a358720]::def_id::DefId, rustc_middle[6b5778e8ec0a50b1]::query::erase::Erased<[u8; 8usize]>>, false, false, false>, rustc_query_impl[9dd290307928cbc3]::plumbing::QueryCtxt>::{closure#0}
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_query_system/src/query/plumbing.rs:481:72
  29:     0x7f9a6b38b7cf - rustc_middle[6b5778e8ec0a50b1]::ty::context::tls::enter_context::<rustc_query_system[67bbb282021fa3e9]::query::plumbing::execute_job_non_incr<rustc_query_impl[9dd290307928cbc3]::DynamicConfig<rustc_query_system[67bbb282021fa3e9]::query::caches::DefaultCache<rustc_span[93bd9c9d5a358720]::def_id::DefId, rustc_middle[6b5778e8ec0a50b1]::query::erase::Erased<[u8; 8usize]>>, false, false, false>, rustc_query_impl[9dd290307928cbc3]::plumbing::QueryCtxt>::{closure#0}, rustc_middle[6b5778e8ec0a50b1]::query::erase::Erased<[u8; 8usize]>>::{closure#0}
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_middle/src/ty/context/tls.rs:82:9
  30:     0x7f9a6b38b7cf - <std[fe8ddff965252baa]::thread::local::LocalKey<core[ad2aefeb2646d5e0]::cell::Cell<*const ()>>>::try_with::<rustc_middle[6b5778e8ec0a50b1]::ty::context::tls::enter_context<rustc_query_system[67bbb282021fa3e9]::query::plumbing::execute_job_non_incr<rustc_query_impl[9dd290307928cbc3]::DynamicConfig<rustc_query_system[67bbb282021fa3e9]::query::caches::DefaultCache<rustc_span[93bd9c9d5a358720]::def_id::DefId, rustc_middle[6b5778e8ec0a50b1]::query::erase::Erased<[u8; 8usize]>>, false, false, false>, rustc_query_impl[9dd290307928cbc3]::plumbing::QueryCtxt>::{closure#0}, rustc_middle[6b5778e8ec0a50b1]::query::erase::Erased<[u8; 8usize]>>::{closure#0}, rustc_middle[6b5778e8ec0a50b1]::query::erase::Erased<[u8; 8usize]>>
                               at /home/matthias/vcs/github/rust_debug_assertions/library/std/src/thread/local.rs:270:16
  31:     0x7f9a6b38b7cf - <std[fe8ddff965252baa]::thread::local::LocalKey<core[ad2aefeb2646d5e0]::cell::Cell<*const ()>>>::with::<rustc_middle[6b5778e8ec0a50b1]::ty::context::tls::enter_context<rustc_query_system[67bbb282021fa3e9]::query::plumbing::execute_job_non_incr<rustc_query_impl[9dd290307928cbc3]::DynamicConfig<rustc_query_system[67bbb282021fa3e9]::query::caches::DefaultCache<rustc_span[93bd9c9d5a358720]::def_id::DefId, rustc_middle[6b5778e8ec0a50b1]::query::erase::Erased<[u8; 8usize]>>, false, false, false>, rustc_query_impl[9dd290307928cbc3]::plumbing::QueryCtxt>::{closure#0}, rustc_middle[6b5778e8ec0a50b1]::query::erase::Erased<[u8; 8usize]>>::{closure#0}, rustc_middle[6b5778e8ec0a50b1]::query::erase::Erased<[u8; 8usize]>>
                               at /home/matthias/vcs/github/rust_debug_assertions/library/std/src/thread/local.rs:246:9
  32:     0x7f9a6b38b7cf - rustc_middle[6b5778e8ec0a50b1]::ty::context::tls::enter_context::<rustc_query_system[67bbb282021fa3e9]::query::plumbing::execute_job_non_incr<rustc_query_impl[9dd290307928cbc3]::DynamicConfig<rustc_query_system[67bbb282021fa3e9]::query::caches::DefaultCache<rustc_span[93bd9c9d5a358720]::def_id::DefId, rustc_middle[6b5778e8ec0a50b1]::query::erase::Erased<[u8; 8usize]>>, false, false, false>, rustc_query_impl[9dd290307928cbc3]::plumbing::QueryCtxt>::{closure#0}, rustc_middle[6b5778e8ec0a50b1]::query::erase::Erased<[u8; 8usize]>>
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_middle/src/ty/context/tls.rs:79:9
  33:     0x7f9a6b38b7cf - <rustc_query_impl[9dd290307928cbc3]::plumbing::QueryCtxt as rustc_query_system[67bbb282021fa3e9]::query::QueryContext>::start_query::<rustc_middle[6b5778e8ec0a50b1]::query::erase::Erased<[u8; 8usize]>, rustc_query_system[67bbb282021fa3e9]::query::plumbing::execute_job_non_incr<rustc_query_impl[9dd290307928cbc3]::DynamicConfig<rustc_query_system[67bbb282021fa3e9]::query::caches::DefaultCache<rustc_span[93bd9c9d5a358720]::def_id::DefId, rustc_middle[6b5778e8ec0a50b1]::query::erase::Erased<[u8; 8usize]>>, false, false, false>, rustc_query_impl[9dd290307928cbc3]::plumbing::QueryCtxt>::{closure#0}>::{closure#0}
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_query_impl/src/plumbing.rs:149:13
  34:     0x7f9a6b38b7cf - rustc_middle[6b5778e8ec0a50b1]::ty::context::tls::with_related_context::<<rustc_query_impl[9dd290307928cbc3]::plumbing::QueryCtxt as rustc_query_system[67bbb282021fa3e9]::query::QueryContext>::start_query<rustc_middle[6b5778e8ec0a50b1]::query::erase::Erased<[u8; 8usize]>, rustc_query_system[67bbb282021fa3e9]::query::plumbing::execute_job_non_incr<rustc_query_impl[9dd290307928cbc3]::DynamicConfig<rustc_query_system[67bbb282021fa3e9]::query::caches::DefaultCache<rustc_span[93bd9c9d5a358720]::def_id::DefId, rustc_middle[6b5778e8ec0a50b1]::query::erase::Erased<[u8; 8usize]>>, false, false, false>, rustc_query_impl[9dd290307928cbc3]::plumbing::QueryCtxt>::{closure#0}>::{closure#0}, rustc_middle[6b5778e8ec0a50b1]::query::erase::Erased<[u8; 8usize]>>::{closure#0}
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_middle/src/ty/context/tls.rs:133:9
  35:     0x7f9a6b38b7cf - rustc_middle[6b5778e8ec0a50b1]::ty::context::tls::with_context::<rustc_middle[6b5778e8ec0a50b1]::ty::context::tls::with_related_context<<rustc_query_impl[9dd290307928cbc3]::plumbing::QueryCtxt as rustc_query_system[67bbb282021fa3e9]::query::QueryContext>::start_query<rustc_middle[6b5778e8ec0a50b1]::query::erase::Erased<[u8; 8usize]>, rustc_query_system[67bbb282021fa3e9]::query::plumbing::execute_job_non_incr<rustc_query_impl[9dd290307928cbc3]::DynamicConfig<rustc_query_system[67bbb282021fa3e9]::query::caches::DefaultCache<rustc_span[93bd9c9d5a358720]::def_id::DefId, rustc_middle[6b5778e8ec0a50b1]::query::erase::Erased<[u8; 8usize]>>, false, false, false>, rustc_query_impl[9dd290307928cbc3]::plumbing::QueryCtxt>::{closure#0}>::{closure#0}, rustc_middle[6b5778e8ec0a50b1]::query::erase::Erased<[u8; 8usize]>>::{closure#0}, rustc_middle[6b5778e8ec0a50b1]::query::erase::Erased<[u8; 8usize]>>::{closure#0}
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_middle/src/ty/context/tls.rs:111:36
  36:     0x7f9a6b38b7cf - rustc_middle[6b5778e8ec0a50b1]::ty::context::tls::with_context_opt::<rustc_middle[6b5778e8ec0a50b1]::ty::context::tls::with_context<rustc_middle[6b5778e8ec0a50b1]::ty::context::tls::with_related_context<<rustc_query_impl[9dd290307928cbc3]::plumbing::QueryCtxt as rustc_query_system[67bbb282021fa3e9]::query::QueryContext>::start_query<rustc_middle[6b5778e8ec0a50b1]::query::erase::Erased<[u8; 8usize]>, rustc_query_system[67bbb282021fa3e9]::query::plumbing::execute_job_non_incr<rustc_query_impl[9dd290307928cbc3]::DynamicConfig<rustc_query_system[67bbb282021fa3e9]::query::caches::DefaultCache<rustc_span[93bd9c9d5a358720]::def_id::DefId, rustc_middle[6b5778e8ec0a50b1]::query::erase::Erased<[u8; 8usize]>>, false, false, false>, rustc_query_impl[9dd290307928cbc3]::plumbing::QueryCtxt>::{closure#0}>::{closure#0}, rustc_middle[6b5778e8ec0a50b1]::query::erase::Erased<[u8; 8usize]>>::{closure#0}, rustc_middle[6b5778e8ec0a50b1]::query::erase::Erased<[u8; 8usize]>>::{closure#0}, rustc_middle[6b5778e8ec0a50b1]::query::erase::Erased<[u8; 8usize]>>
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_middle/src/ty/context/tls.rs:100:18
  37:     0x7f9a6b38b7cf - rustc_middle[6b5778e8ec0a50b1]::ty::context::tls::with_context::<rustc_middle[6b5778e8ec0a50b1]::ty::context::tls::with_related_context<<rustc_query_impl[9dd290307928cbc3]::plumbing::QueryCtxt as rustc_query_system[67bbb282021fa3e9]::query::QueryContext>::start_query<rustc_middle[6b5778e8ec0a50b1]::query::erase::Erased<[u8; 8usize]>, rustc_query_system[67bbb282021fa3e9]::query::plumbing::execute_job_non_incr<rustc_query_impl[9dd290307928cbc3]::DynamicConfig<rustc_query_system[67bbb282021fa3e9]::query::caches::DefaultCache<rustc_span[93bd9c9d5a358720]::def_id::DefId, rustc_middle[6b5778e8ec0a50b1]::query::erase::Erased<[u8; 8usize]>>, false, false, false>, rustc_query_impl[9dd290307928cbc3]::plumbing::QueryCtxt>::{closure#0}>::{closure#0}, rustc_middle[6b5778e8ec0a50b1]::query::erase::Erased<[u8; 8usize]>>::{closure#0}, rustc_middle[6b5778e8ec0a50b1]::query::erase::Erased<[u8; 8usize]>>
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_middle/src/ty/context/tls.rs:111:5
  38:     0x7f9a6b38b7cf - rustc_middle[6b5778e8ec0a50b1]::ty::context::tls::with_related_context::<<rustc_query_impl[9dd290307928cbc3]::plumbing::QueryCtxt as rustc_query_system[67bbb282021fa3e9]::query::QueryContext>::start_query<rustc_middle[6b5778e8ec0a50b1]::query::erase::Erased<[u8; 8usize]>, rustc_query_system[67bbb282021fa3e9]::query::plumbing::execute_job_non_incr<rustc_query_impl[9dd290307928cbc3]::DynamicConfig<rustc_query_system[67bbb282021fa3e9]::query::caches::DefaultCache<rustc_span[93bd9c9d5a358720]::def_id::DefId, rustc_middle[6b5778e8ec0a50b1]::query::erase::Erased<[u8; 8usize]>>, false, false, false>, rustc_query_impl[9dd290307928cbc3]::plumbing::QueryCtxt>::{closure#0}>::{closure#0}, rustc_middle[6b5778e8ec0a50b1]::query::erase::Erased<[u8; 8usize]>>
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_middle/src/ty/context/tls.rs:124:5
  39:     0x7f9a6b38b7cf - <rustc_query_impl[9dd290307928cbc3]::plumbing::QueryCtxt as rustc_query_system[67bbb282021fa3e9]::query::QueryContext>::start_query::<rustc_middle[6b5778e8ec0a50b1]::query::erase::Erased<[u8; 8usize]>, rustc_query_system[67bbb282021fa3e9]::query::plumbing::execute_job_non_incr<rustc_query_impl[9dd290307928cbc3]::DynamicConfig<rustc_query_system[67bbb282021fa3e9]::query::caches::DefaultCache<rustc_span[93bd9c9d5a358720]::def_id::DefId, rustc_middle[6b5778e8ec0a50b1]::query::erase::Erased<[u8; 8usize]>>, false, false, false>, rustc_query_impl[9dd290307928cbc3]::plumbing::QueryCtxt>::{closure#0}>
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_query_impl/src/plumbing.rs:134:9
  40:     0x7f9a6b38b7cf - rustc_query_system[67bbb282021fa3e9]::query::plumbing::execute_job_non_incr::<rustc_query_impl[9dd290307928cbc3]::DynamicConfig<rustc_query_system[67bbb282021fa3e9]::query::caches::DefaultCache<rustc_span[93bd9c9d5a358720]::def_id::DefId, rustc_middle[6b5778e8ec0a50b1]::query::erase::Erased<[u8; 8usize]>>, false, false, false>, rustc_query_impl[9dd290307928cbc3]::plumbing::QueryCtxt>
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_query_system/src/query/plumbing.rs:481:18
  41:     0x7f9a6b38b7cf - rustc_query_system[67bbb282021fa3e9]::query::plumbing::execute_job::<rustc_query_impl[9dd290307928cbc3]::DynamicConfig<rustc_query_system[67bbb282021fa3e9]::query::caches::DefaultCache<rustc_span[93bd9c9d5a358720]::def_id::DefId, rustc_middle[6b5778e8ec0a50b1]::query::erase::Erased<[u8; 8usize]>>, false, false, false>, rustc_query_impl[9dd290307928cbc3]::plumbing::QueryCtxt, false>
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_query_system/src/query/plumbing.rs:414:9
  42:     0x7f9a6b38b7cf - rustc_query_system[67bbb282021fa3e9]::query::plumbing::try_execute_query::<rustc_query_impl[9dd290307928cbc3]::DynamicConfig<rustc_query_system[67bbb282021fa3e9]::query::caches::DefaultCache<rustc_span[93bd9c9d5a358720]::def_id::DefId, rustc_middle[6b5778e8ec0a50b1]::query::erase::Erased<[u8; 8usize]>>, false, false, false>, rustc_query_impl[9dd290307928cbc3]::plumbing::QueryCtxt, false>
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_query_system/src/query/plumbing.rs:359:13
  43:     0x7f9a6b48353a - rustc_query_system[67bbb282021fa3e9]::query::plumbing::get_query_non_incr::<rustc_query_impl[9dd290307928cbc3]::DynamicConfig<rustc_query_system[67bbb282021fa3e9]::query::caches::DefaultCache<rustc_span[93bd9c9d5a358720]::def_id::DefId, rustc_middle[6b5778e8ec0a50b1]::query::erase::Erased<[u8; 8usize]>>, false, false, false>, rustc_query_impl[9dd290307928cbc3]::plumbing::QueryCtxt>::{closure#0}
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_query_system/src/query/plumbing.rs:811:32
  44:     0x7f9a6b48353a - stacker[7116c11b72288e56]::maybe_grow::<rustc_middle[6b5778e8ec0a50b1]::query::erase::Erased<[u8; 8usize]>, rustc_query_system[67bbb282021fa3e9]::query::plumbing::get_query_non_incr<rustc_query_impl[9dd290307928cbc3]::DynamicConfig<rustc_query_system[67bbb282021fa3e9]::query::caches::DefaultCache<rustc_span[93bd9c9d5a358720]::def_id::DefId, rustc_middle[6b5778e8ec0a50b1]::query::erase::Erased<[u8; 8usize]>>, false, false, false>, rustc_query_impl[9dd290307928cbc3]::plumbing::QueryCtxt>::{closure#0}>
                               at /home/matthias/.cargo/registry/src/index.crates.io-6f17d22bba15001f/stacker-0.1.15/src/lib.rs:55:9
  45:     0x7f9a6b48353a - rustc_data_structures[b8a0954bf6a11949]::stack::ensure_sufficient_stack::<rustc_middle[6b5778e8ec0a50b1]::query::erase::Erased<[u8; 8usize]>, rustc_query_system[67bbb282021fa3e9]::query::plumbing::get_query_non_incr<rustc_query_impl[9dd290307928cbc3]::DynamicConfig<rustc_query_system[67bbb282021fa3e9]::query::caches::DefaultCache<rustc_span[93bd9c9d5a358720]::def_id::DefId, rustc_middle[6b5778e8ec0a50b1]::query::erase::Erased<[u8; 8usize]>>, false, false, false>, rustc_query_impl[9dd290307928cbc3]::plumbing::QueryCtxt>::{closure#0}>
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_data_structures/src/stack.rs:17:5
  46:     0x7f9a6b48353a - rustc_query_system[67bbb282021fa3e9]::query::plumbing::get_query_non_incr::<rustc_query_impl[9dd290307928cbc3]::DynamicConfig<rustc_query_system[67bbb282021fa3e9]::query::caches::DefaultCache<rustc_span[93bd9c9d5a358720]::def_id::DefId, rustc_middle[6b5778e8ec0a50b1]::query::erase::Erased<[u8; 8usize]>>, false, false, false>, rustc_query_impl[9dd290307928cbc3]::plumbing::QueryCtxt>
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_query_system/src/query/plumbing.rs:811:5
  47:     0x7f9a6b48353a - rustc_query_impl[9dd290307928cbc3]::query_impl::collect_return_position_impl_trait_in_trait_tys::get_query_non_incr::__rust_end_short_backtrace
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_query_impl/src/plumbing.rs:556:26
  48:     0x7f9a69ccce03 - rustc_middle[6b5778e8ec0a50b1]::query::plumbing::query_get_at::<rustc_query_system[67bbb282021fa3e9]::query::caches::DefaultCache<rustc_span[93bd9c9d5a358720]::def_id::DefId, rustc_middle[6b5778e8ec0a50b1]::query::erase::Erased<[u8; 8usize]>>>
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_middle/src/query/plumbing.rs:155:17
  49:     0x7f9a69ce17b5 - <rustc_middle[6b5778e8ec0a50b1]::query::plumbing::TyCtxtAt>::collect_return_position_impl_trait_in_trait_tys::<rustc_span[93bd9c9d5a358720]::def_id::DefId>
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_middle/src/query/plumbing.rs:385:31
  50:     0x7f9a69ce17b5 - <rustc_middle[6b5778e8ec0a50b1]::ty::context::TyCtxt>::collect_return_position_impl_trait_in_trait_tys::<rustc_span[93bd9c9d5a358720]::def_id::DefId>
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_middle/src/query/plumbing.rs:376:17
  51:     0x7f9a69ce17b5 - rustc_hir_analysis[38d60f9cb4d84cc8]::collect::type_of::type_of
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_hir_analysis/src/collect/type_of.rs:314:15
  52:     0x7f9a6b42d815 - rustc_query_impl[9dd290307928cbc3]::query_impl::type_of::dynamic_query::{closure#2}::{closure#0}
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_query_impl/src/plumbing.rs:582:47
  53:     0x7f9a6b42d815 - rustc_query_impl[9dd290307928cbc3]::plumbing::__rust_begin_short_backtrace::<rustc_query_impl[9dd290307928cbc3]::query_impl::type_of::dynamic_query::{closure#2}::{closure#0}, rustc_middle[6b5778e8ec0a50b1]::query::erase::Erased<[u8; 8usize]>>
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_query_impl/src/plumbing.rs:506:18
  54:     0x7f9a6b5f55df - rustc_query_impl[9dd290307928cbc3]::query_impl::type_of::dynamic_query::{closure#2}
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_query_impl/src/plumbing.rs:578:25
  55:     0x7f9a6b5f55df - <rustc_query_impl[9dd290307928cbc3]::query_impl::type_of::dynamic_query::{closure#2} as core[ad2aefeb2646d5e0]::ops::function::FnOnce<(rustc_middle[6b5778e8ec0a50b1]::ty::context::TyCtxt, rustc_span[93bd9c9d5a358720]::def_id::DefId)>>::call_once
                               at /home/matthias/vcs/github/rust_debug_assertions/library/core/src/ops/function.rs:250:5
  56:     0x7f9a6b38b7cf - <rustc_query_impl[9dd290307928cbc3]::DynamicConfig<rustc_query_system[67bbb282021fa3e9]::query::caches::DefaultCache<rustc_span[93bd9c9d5a358720]::def_id::DefId, rustc_middle[6b5778e8ec0a50b1]::query::erase::Erased<[u8; 8usize]>>, false, false, false> as rustc_query_system[67bbb282021fa3e9]::query::config::QueryConfig<rustc_query_impl[9dd290307928cbc3]::plumbing::QueryCtxt>>::compute
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_query_impl/src/lib.rs:116:9
  57:     0x7f9a6b38b7cf - rustc_query_system[67bbb282021fa3e9]::query::plumbing::execute_job_non_incr::<rustc_query_impl[9dd290307928cbc3]::DynamicConfig<rustc_query_system[67bbb282021fa3e9]::query::caches::DefaultCache<rustc_span[93bd9c9d5a358720]::def_id::DefId, rustc_middle[6b5778e8ec0a50b1]::query::erase::Erased<[u8; 8usize]>>, false, false, false>, rustc_query_impl[9dd290307928cbc3]::plumbing::QueryCtxt>::{closure#0}
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_query_system/src/query/plumbing.rs:481:72
  58:     0x7f9a6b38b7cf - rustc_middle[6b5778e8ec0a50b1]::ty::context::tls::enter_context::<rustc_query_system[67bbb282021fa3e9]::query::plumbing::execute_job_non_incr<rustc_query_impl[9dd290307928cbc3]::DynamicConfig<rustc_query_system[67bbb282021fa3e9]::query::caches::DefaultCache<rustc_span[93bd9c9d5a358720]::def_id::DefId, rustc_middle[6b5778e8ec0a50b1]::query::erase::Erased<[u8; 8usize]>>, false, false, false>, rustc_query_impl[9dd290307928cbc3]::plumbing::QueryCtxt>::{closure#0}, rustc_middle[6b5778e8ec0a50b1]::query::erase::Erased<[u8; 8usize]>>::{closure#0}
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_middle/src/ty/context/tls.rs:82:9
  59:     0x7f9a6b38b7cf - <std[fe8ddff965252baa]::thread::local::LocalKey<core[ad2aefeb2646d5e0]::cell::Cell<*const ()>>>::try_with::<rustc_middle[6b5778e8ec0a50b1]::ty::context::tls::enter_context<rustc_query_system[67bbb282021fa3e9]::query::plumbing::execute_job_non_incr<rustc_query_impl[9dd290307928cbc3]::DynamicConfig<rustc_query_system[67bbb282021fa3e9]::query::caches::DefaultCache<rustc_span[93bd9c9d5a358720]::def_id::DefId, rustc_middle[6b5778e8ec0a50b1]::query::erase::Erased<[u8; 8usize]>>, false, false, false>, rustc_query_impl[9dd290307928cbc3]::plumbing::QueryCtxt>::{closure#0}, rustc_middle[6b5778e8ec0a50b1]::query::erase::Erased<[u8; 8usize]>>::{closure#0}, rustc_middle[6b5778e8ec0a50b1]::query::erase::Erased<[u8; 8usize]>>
                               at /home/matthias/vcs/github/rust_debug_assertions/library/std/src/thread/local.rs:270:16
  60:     0x7f9a6b38b7cf - <std[fe8ddff965252baa]::thread::local::LocalKey<core[ad2aefeb2646d5e0]::cell::Cell<*const ()>>>::with::<rustc_middle[6b5778e8ec0a50b1]::ty::context::tls::enter_context<rustc_query_system[67bbb282021fa3e9]::query::plumbing::execute_job_non_incr<rustc_query_impl[9dd290307928cbc3]::DynamicConfig<rustc_query_system[67bbb282021fa3e9]::query::caches::DefaultCache<rustc_span[93bd9c9d5a358720]::def_id::DefId, rustc_middle[6b5778e8ec0a50b1]::query::erase::Erased<[u8; 8usize]>>, false, false, false>, rustc_query_impl[9dd290307928cbc3]::plumbing::QueryCtxt>::{closure#0}, rustc_middle[6b5778e8ec0a50b1]::query::erase::Erased<[u8; 8usize]>>::{closure#0}, rustc_middle[6b5778e8ec0a50b1]::query::erase::Erased<[u8; 8usize]>>
                               at /home/matthias/vcs/github/rust_debug_assertions/library/std/src/thread/local.rs:246:9
  61:     0x7f9a6b38b7cf - rustc_middle[6b5778e8ec0a50b1]::ty::context::tls::enter_context::<rustc_query_system[67bbb282021fa3e9]::query::plumbing::execute_job_non_incr<rustc_query_impl[9dd290307928cbc3]::DynamicConfig<rustc_query_system[67bbb282021fa3e9]::query::caches::DefaultCache<rustc_span[93bd9c9d5a358720]::def_id::DefId, rustc_middle[6b5778e8ec0a50b1]::query::erase::Erased<[u8; 8usize]>>, false, false, false>, rustc_query_impl[9dd290307928cbc3]::plumbing::QueryCtxt>::{closure#0}, rustc_middle[6b5778e8ec0a50b1]::query::erase::Erased<[u8; 8usize]>>
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_middle/src/ty/context/tls.rs:79:9
  62:     0x7f9a6b38b7cf - <rustc_query_impl[9dd290307928cbc3]::plumbing::QueryCtxt as rustc_query_system[67bbb282021fa3e9]::query::QueryContext>::start_query::<rustc_middle[6b5778e8ec0a50b1]::query::erase::Erased<[u8; 8usize]>, rustc_query_system[67bbb282021fa3e9]::query::plumbing::execute_job_non_incr<rustc_query_impl[9dd290307928cbc3]::DynamicConfig<rustc_query_system[67bbb282021fa3e9]::query::caches::DefaultCache<rustc_span[93bd9c9d5a358720]::def_id::DefId, rustc_middle[6b5778e8ec0a50b1]::query::erase::Erased<[u8; 8usize]>>, false, false, false>, rustc_query_impl[9dd290307928cbc3]::plumbing::QueryCtxt>::{closure#0}>::{closure#0}
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_query_impl/src/plumbing.rs:149:13
  63:     0x7f9a6b38b7cf - rustc_middle[6b5778e8ec0a50b1]::ty::context::tls::with_related_context::<<rustc_query_impl[9dd290307928cbc3]::plumbing::QueryCtxt as rustc_query_system[67bbb282021fa3e9]::query::QueryContext>::start_query<rustc_middle[6b5778e8ec0a50b1]::query::erase::Erased<[u8; 8usize]>, rustc_query_system[67bbb282021fa3e9]::query::plumbing::execute_job_non_incr<rustc_query_impl[9dd290307928cbc3]::DynamicConfig<rustc_query_system[67bbb282021fa3e9]::query::caches::DefaultCache<rustc_span[93bd9c9d5a358720]::def_id::DefId, rustc_middle[6b5778e8ec0a50b1]::query::erase::Erased<[u8; 8usize]>>, false, false, false>, rustc_query_impl[9dd290307928cbc3]::plumbing::QueryCtxt>::{closure#0}>::{closure#0}, rustc_middle[6b5778e8ec0a50b1]::query::erase::Erased<[u8; 8usize]>>::{closure#0}
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_middle/src/ty/context/tls.rs:133:9
  64:     0x7f9a6b38b7cf - rustc_middle[6b5778e8ec0a50b1]::ty::context::tls::with_context::<rustc_middle[6b5778e8ec0a50b1]::ty::context::tls::with_related_context<<rustc_query_impl[9dd290307928cbc3]::plumbing::QueryCtxt as rustc_query_system[67bbb282021fa3e9]::query::QueryContext>::start_query<rustc_middle[6b5778e8ec0a50b1]::query::erase::Erased<[u8; 8usize]>, rustc_query_system[67bbb282021fa3e9]::query::plumbing::execute_job_non_incr<rustc_query_impl[9dd290307928cbc3]::DynamicConfig<rustc_query_system[67bbb282021fa3e9]::query::caches::DefaultCache<rustc_span[93bd9c9d5a358720]::def_id::DefId, rustc_middle[6b5778e8ec0a50b1]::query::erase::Erased<[u8; 8usize]>>, false, false, false>, rustc_query_impl[9dd290307928cbc3]::plumbing::QueryCtxt>::{closure#0}>::{closure#0}, rustc_middle[6b5778e8ec0a50b1]::query::erase::Erased<[u8; 8usize]>>::{closure#0}, rustc_middle[6b5778e8ec0a50b1]::query::erase::Erased<[u8; 8usize]>>::{closure#0}
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_middle/src/ty/context/tls.rs:111:36
  65:     0x7f9a6b38b7cf - rustc_middle[6b5778e8ec0a50b1]::ty::context::tls::with_context_opt::<rustc_middle[6b5778e8ec0a50b1]::ty::context::tls::with_context<rustc_middle[6b5778e8ec0a50b1]::ty::context::tls::with_related_context<<rustc_query_impl[9dd290307928cbc3]::plumbing::QueryCtxt as rustc_query_system[67bbb282021fa3e9]::query::QueryContext>::start_query<rustc_middle[6b5778e8ec0a50b1]::query::erase::Erased<[u8; 8usize]>, rustc_query_system[67bbb282021fa3e9]::query::plumbing::execute_job_non_incr<rustc_query_impl[9dd290307928cbc3]::DynamicConfig<rustc_query_system[67bbb282021fa3e9]::query::caches::DefaultCache<rustc_span[93bd9c9d5a358720]::def_id::DefId, rustc_middle[6b5778e8ec0a50b1]::query::erase::Erased<[u8; 8usize]>>, false, false, false>, rustc_query_impl[9dd290307928cbc3]::plumbing::QueryCtxt>::{closure#0}>::{closure#0}, rustc_middle[6b5778e8ec0a50b1]::query::erase::Erased<[u8; 8usize]>>::{closure#0}, rustc_middle[6b5778e8ec0a50b1]::query::erase::Erased<[u8; 8usize]>>::{closure#0}, rustc_middle[6b5778e8ec0a50b1]::query::erase::Erased<[u8; 8usize]>>
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_middle/src/ty/context/tls.rs:100:18
  66:     0x7f9a6b38b7cf - rustc_middle[6b5778e8ec0a50b1]::ty::context::tls::with_context::<rustc_middle[6b5778e8ec0a50b1]::ty::context::tls::with_related_context<<rustc_query_impl[9dd290307928cbc3]::plumbing::QueryCtxt as rustc_query_system[67bbb282021fa3e9]::query::QueryContext>::start_query<rustc_middle[6b5778e8ec0a50b1]::query::erase::Erased<[u8; 8usize]>, rustc_query_system[67bbb282021fa3e9]::query::plumbing::execute_job_non_incr<rustc_query_impl[9dd290307928cbc3]::DynamicConfig<rustc_query_system[67bbb282021fa3e9]::query::caches::DefaultCache<rustc_span[93bd9c9d5a358720]::def_id::DefId, rustc_middle[6b5778e8ec0a50b1]::query::erase::Erased<[u8; 8usize]>>, false, false, false>, rustc_query_impl[9dd290307928cbc3]::plumbing::QueryCtxt>::{closure#0}>::{closure#0}, rustc_middle[6b5778e8ec0a50b1]::query::erase::Erased<[u8; 8usize]>>::{closure#0}, rustc_middle[6b5778e8ec0a50b1]::query::erase::Erased<[u8; 8usize]>>
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_middle/src/ty/context/tls.rs:111:5
  67:     0x7f9a6b38b7cf - rustc_middle[6b5778e8ec0a50b1]::ty::context::tls::with_related_context::<<rustc_query_impl[9dd290307928cbc3]::plumbing::QueryCtxt as rustc_query_system[67bbb282021fa3e9]::query::QueryContext>::start_query<rustc_middle[6b5778e8ec0a50b1]::query::erase::Erased<[u8; 8usize]>, rustc_query_system[67bbb282021fa3e9]::query::plumbing::execute_job_non_incr<rustc_query_impl[9dd290307928cbc3]::DynamicConfig<rustc_query_system[67bbb282021fa3e9]::query::caches::DefaultCache<rustc_span[93bd9c9d5a358720]::def_id::DefId, rustc_middle[6b5778e8ec0a50b1]::query::erase::Erased<[u8; 8usize]>>, false, false, false>, rustc_query_impl[9dd290307928cbc3]::plumbing::QueryCtxt>::{closure#0}>::{closure#0}, rustc_middle[6b5778e8ec0a50b1]::query::erase::Erased<[u8; 8usize]>>
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_middle/src/ty/context/tls.rs:124:5
  68:     0x7f9a6b38b7cf - <rustc_query_impl[9dd290307928cbc3]::plumbing::QueryCtxt as rustc_query_system[67bbb282021fa3e9]::query::QueryContext>::start_query::<rustc_middle[6b5778e8ec0a50b1]::query::erase::Erased<[u8; 8usize]>, rustc_query_system[67bbb282021fa3e9]::query::plumbing::execute_job_non_incr<rustc_query_impl[9dd290307928cbc3]::DynamicConfig<rustc_query_system[67bbb282021fa3e9]::query::caches::DefaultCache<rustc_span[93bd9c9d5a358720]::def_id::DefId, rustc_middle[6b5778e8ec0a50b1]::query::erase::Erased<[u8; 8usize]>>, false, false, false>, rustc_query_impl[9dd290307928cbc3]::plumbing::QueryCtxt>::{closure#0}>
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_query_impl/src/plumbing.rs:134:9
  69:     0x7f9a6b38b7cf - rustc_query_system[67bbb282021fa3e9]::query::plumbing::execute_job_non_incr::<rustc_query_impl[9dd290307928cbc3]::DynamicConfig<rustc_query_system[67bbb282021fa3e9]::query::caches::DefaultCache<rustc_span[93bd9c9d5a358720]::def_id::DefId, rustc_middle[6b5778e8ec0a50b1]::query::erase::Erased<[u8; 8usize]>>, false, false, false>, rustc_query_impl[9dd290307928cbc3]::plumbing::QueryCtxt>
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_query_system/src/query/plumbing.rs:481:18
  70:     0x7f9a6b38b7cf - rustc_query_system[67bbb282021fa3e9]::query::plumbing::execute_job::<rustc_query_impl[9dd290307928cbc3]::DynamicConfig<rustc_query_system[67bbb282021fa3e9]::query::caches::DefaultCache<rustc_span[93bd9c9d5a358720]::def_id::DefId, rustc_middle[6b5778e8ec0a50b1]::query::erase::Erased<[u8; 8usize]>>, false, false, false>, rustc_query_impl[9dd290307928cbc3]::plumbing::QueryCtxt, false>
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_query_system/src/query/plumbing.rs:414:9
  71:     0x7f9a6b38b7cf - rustc_query_system[67bbb282021fa3e9]::query::plumbing::try_execute_query::<rustc_query_impl[9dd290307928cbc3]::DynamicConfig<rustc_query_system[67bbb282021fa3e9]::query::caches::DefaultCache<rustc_span[93bd9c9d5a358720]::def_id::DefId, rustc_middle[6b5778e8ec0a50b1]::query::erase::Erased<[u8; 8usize]>>, false, false, false>, rustc_query_impl[9dd290307928cbc3]::plumbing::QueryCtxt, false>
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_query_system/src/query/plumbing.rs:359:13
  72:     0x7f9a6b5f7b0a - rustc_query_system[67bbb282021fa3e9]::query::plumbing::get_query_non_incr::<rustc_query_impl[9dd290307928cbc3]::DynamicConfig<rustc_query_system[67bbb282021fa3e9]::query::caches::DefaultCache<rustc_span[93bd9c9d5a358720]::def_id::DefId, rustc_middle[6b5778e8ec0a50b1]::query::erase::Erased<[u8; 8usize]>>, false, false, false>, rustc_query_impl[9dd290307928cbc3]::plumbing::QueryCtxt>::{closure#0}
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_query_system/src/query/plumbing.rs:811:32
  73:     0x7f9a6b5f7b0a - stacker[7116c11b72288e56]::maybe_grow::<rustc_middle[6b5778e8ec0a50b1]::query::erase::Erased<[u8; 8usize]>, rustc_query_system[67bbb282021fa3e9]::query::plumbing::get_query_non_incr<rustc_query_impl[9dd290307928cbc3]::DynamicConfig<rustc_query_system[67bbb282021fa3e9]::query::caches::DefaultCache<rustc_span[93bd9c9d5a358720]::def_id::DefId, rustc_middle[6b5778e8ec0a50b1]::query::erase::Erased<[u8; 8usize]>>, false, false, false>, rustc_query_impl[9dd290307928cbc3]::plumbing::QueryCtxt>::{closure#0}>
                               at /home/matthias/.cargo/registry/src/index.crates.io-6f17d22bba15001f/stacker-0.1.15/src/lib.rs:55:9
  74:     0x7f9a6b5f7b0a - rustc_data_structures[b8a0954bf6a11949]::stack::ensure_sufficient_stack::<rustc_middle[6b5778e8ec0a50b1]::query::erase::Erased<[u8; 8usize]>, rustc_query_system[67bbb282021fa3e9]::query::plumbing::get_query_non_incr<rustc_query_impl[9dd290307928cbc3]::DynamicConfig<rustc_query_system[67bbb282021fa3e9]::query::caches::DefaultCache<rustc_span[93bd9c9d5a358720]::def_id::DefId, rustc_middle[6b5778e8ec0a50b1]::query::erase::Erased<[u8; 8usize]>>, false, false, false>, rustc_query_impl[9dd290307928cbc3]::plumbing::QueryCtxt>::{closure#0}>
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_data_structures/src/stack.rs:17:5
  75:     0x7f9a6b5f7b0a - rustc_query_system[67bbb282021fa3e9]::query::plumbing::get_query_non_incr::<rustc_query_impl[9dd290307928cbc3]::DynamicConfig<rustc_query_system[67bbb282021fa3e9]::query::caches::DefaultCache<rustc_span[93bd9c9d5a358720]::def_id::DefId, rustc_middle[6b5778e8ec0a50b1]::query::erase::Erased<[u8; 8usize]>>, false, false, false>, rustc_query_impl[9dd290307928cbc3]::plumbing::QueryCtxt>
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_query_system/src/query/plumbing.rs:811:5
  76:     0x7f9a6b5f7b0a - rustc_query_impl[9dd290307928cbc3]::query_impl::type_of::get_query_non_incr::__rust_end_short_backtrace
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_query_impl/src/plumbing.rs:556:26
  77:     0x7f9a69d3ab33 - rustc_middle[6b5778e8ec0a50b1]::query::plumbing::query_get_at::<rustc_query_system[67bbb282021fa3e9]::query::caches::DefaultCache<rustc_span[93bd9c9d5a358720]::def_id::DefId, rustc_middle[6b5778e8ec0a50b1]::query::erase::Erased<[u8; 8usize]>>>
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_middle/src/query/plumbing.rs:155:17
  78:     0x7f9a69d3f65f - <rustc_middle[6b5778e8ec0a50b1]::query::plumbing::TyCtxtAt>::type_of::<&rustc_span[93bd9c9d5a358720]::def_id::DefId>
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_middle/src/query/plumbing.rs:385:31
  79:     0x7f9a69d3f65f - <rustc_middle[6b5778e8ec0a50b1]::ty::context::TyCtxt>::type_of::<&rustc_span[93bd9c9d5a358720]::def_id::DefId>
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_middle/src/query/plumbing.rs:376:17
  80:     0x7f9a69d3f65f - rustc_hir_analysis[38d60f9cb4d84cc8]::impl_wf_check::enforce_impl_params_are_constrained::{closure#0}
                               at /home/matthias/vcs/github/rust_debug_assertions/compiler/rustc_hir_analysis/src/impl_wf_check.rs:107:46
  81:     0x7f9a69d3f65f - core[ad2aefeb2646d5e0]::iter::adapters::map::map_fold::<&rustc_span[93bd9c9d5a358720]::def_id::DefId, alloc[59fef786b468a15e]::vec::Vec<rustc_hir_analysis[38d60f9cb4d84cc8]::constrained_generic_params::Parameter>, (), rustc_hir_analysis[38d60f9cb4d84cc8]::impl_wf_check::enforce_impl_params_are_constrained::{closure#0}, <core[ad2aefeb2646d5e0]::iter::adapters::flatten::FlattenCompat<_, _>>::iter_fold::flatten<alloc[59fef786b468a15e]::vec::Vec<rustc_hir_analysis[38d60f9cb4d84cc8]::constrained_generic_params::Parameter>, (), <core[ad2aefeb2646d5e0]::iter::adapters::flatten::FlattenCompat<_, _> as core[ad2aefeb2646d5e0]::iter::traits::iterator::Iterator>::fold::flatten<alloc[59fef786b468a15e]::vec::into_iter::IntoIter<rustc_hir_analysis[38d60f9cb4d84cc8]::constrained_generic_params::Parameter>, (), core[ad2aefeb2646d5e0]::iter::adapters::map::map_fold<rustc_hir_analysis[38d60f9cb4d84cc8]::constrained_generic_params::Parameter, (rustc_hir_analysis[38d60f9cb4d84cc8]::constrained_generic_params::Parameter, ()), (), <hashbrown[26bc3f9f7bedacb6]::set::HashSet<rustc_hir_analysis[38d60f9cb4d84cc8]::constrained_generic_params::Parameter, core[ad2aefeb2646d5e0]::hash::BuildHasherDefault<rustc_hash[a52e4748ddd034eb]::FxHasher>> as core[ad2aefeb2646d5e0]::iter::traits::collect::Extend<rustc_hir_analysis[38d60f9cb4d84cc8]::constrained_generic_params::Parameter>>::extend<core[ad2aefeb2646d5e0]::iter::adapters::flatten::FlatMap<core[ad2aefeb2646d5e0]::slice::iter::Iter<rustc_span[93bd9c9d5a358720]::def_id::DefId>, alloc[59fef786b468a15e]::vec::Vec<rustc_hir_analysis[38d60f9cb4d84cc8]::constrained_generic_params::Parameter>, rustc_hir_analysis[38d60f9cb4d84cc8]::impl_wf_check::enforce_impl_params_are_constrained::{closure#0}>>::{closure#0}, core[ad2aefeb2646d5e0]::iter::traits::iterator::Iterator::for_each::call<(rustc_hir_analysis[38d60f9cb4d84cc8]::constrained_generic_params::Parameter, ()), <hashbrown[26bc3f9f7bedacb6]::map::HashMap<rustc_hir_analysis[38d60f9cb4d84cc8]::constrained_generic_params::Parameter, (), core[ad2aefeb2646d5e0]::hash::BuildHasherDefault<rustc_hash[a52e4748ddd034eb]::FxHasher>> as core[ad2aefeb2646d5e0]::iter::traits::collect::Extend<(rustc_hir_analysis[38d60f9cb4d84cc8]::constrained_generic_params::Parameter, ())>>::extend<core[ad2aefeb2646d5e0]::iter::adapters::map::Map<core[ad2aefeb2646d5e0]::iter::adapters::flatten::FlatMap<core[ad2aefeb2646d5e0]::slice::iter::Iter<rustc_span[93bd9c9d5a358720]::def_id::DefId>, alloc[59fef786b468a15e]::vec::Vec<rustc_hir_analysis[38d60f9cb4d84cc8]::constrained_generic_params::Parameter>, rustc_hir_analysis[38d60f9cb4d84cc8]::impl_wf_check::enforce_impl_params_are_constrained::{closure#0}>, <hashbrown[26bc3f9f7bedacb6]::set::HashSet<rustc_hir_analysis[38d60f9cb4d84cc8]::constrained_generic_params::Parameter, core[ad2aefeb2646d5e0]::hash::BuildHasherDefault<rustc_hash[a52e4748ddd034eb]::FxHasher>> as core[ad2aefeb2646d5e0]::iter::traits::collect::Extend<rustc_hir_analysis[38d60f9cb4d84cc8]::constrained_generic_params::Parameter>>::extend<core[ad2aefeb2646d5e0]::iter::adapters::flatten::FlatMap<core[ad2aefeb2646d5e0]::slice::iter::Iter<rustc_span[93bd9c9d5a358720]::def_id::DefId>, alloc[59fef786b468a15e]::vec::Vec<rustc_hir_analysis[38d60f9cb4d84cc8]::constrained_generic_params::Parameter>, rustc_hir_analysis[38d60f9cb4d84cc8]::impl_wf_check::enforce_impl_params_are_constrained::{closure#0}>>::{closure#0}>>::{closure#0}>::{closure#0}>::{closure#0}>::{closure#0}>::{closure#0}>::{closure#0}
                               at /home/matthias/vcs/github/rust_debug_assertions/library/core/src/iter/adapters/map.rs:84:28
  82:     0x7f9a69d3f65f - <core[ad2aefeb2646d5e0]::slice::iter::Iter<rustc_span[93bd9c9d5a358720]::def_id::DefId> as core[ad2aefeb2646d5e0]::iter::traits::iterator::Iterator>::fold::<(), core[ad2aefeb2646d5e0]::iter::adapters::map::map_fold<&rustc_span[93bd9c9d5a358720]::def_id::DefId, alloc[59fef786b468a15e]::vec::Vec<rustc_hir_analysis[38d60f9cb4d84cc8]::constrained_generic_params::Parameter>, (), rustc_hir_analysis[38d60f9cb4d84cc8]::impl_wf_check::enforce_impl_params_are_constrained::{closure#0}, <core[ad2aefeb2646d5e0]::iter::adapters::flatten::FlattenCompat<_, _>>::iter_fold::flatten<alloc[59fef786b468a15e]::vec::Vec<rustc_hir_analysis[38d60f9cb4d84cc8]::constrained_generic_params::Parameter>, (), <core[ad2aefeb2646d5e0]::iter::adapters::flatten::FlattenCompat<_, _> as core[ad2aefeb2646d5e0]::iter::traits::iterator::Iterator>::fold::flatten<alloc[59fef786b468a15e]::vec::into_iter::IntoIter<rustc_hir_analysis[38d60f9cb4d84cc8]::constrained_generic_params::Parameter>, (), core[ad2aefeb2646d5e0]::iter::adapters::map::map_fold<rustc_hir_analysis[38d60f9cb4d84cc8]::constrained_generic_params::Parameter, (rustc_hir_analysis[38d60f9cb4d84cc8]::constrained_generic_params::Parameter, ()), (), <hashbrown[26bc3f9f7bedacb6]::set::HashSet<rustc_hir_analysis[38d60f9cb4d84cc8]::constrained_generic_params::Parameter, core[ad2aefeb2646d5e0]::hash::BuildHasherDefault<rustc_hash[a52e4748ddd034eb]::FxHasher>> as core[ad2aefeb2646d5e0]::iter::traits::collect::Extend<rustc_hir_analysis[38d60f9cb4d84cc8]::constrained_generic_params::Parameter>>::extend<core[ad2aefeb2646d5e0]::iter::adapters::flatten::FlatMap<core[ad2aefeb2646d5e0]::slice::iter::Iter<rustc_span[93bd9c9d5a358720]::def_id::DefId>, alloc[59fef786b468a15e]::vec::Vec<rustc_hir_analysis[38d60f9cb4d84cc8]::constrained_generic_params::Parameter>, rustc_hir_analysis[38d60f9cb4d84cc8]::impl_wf_check::enforce_impl_params_are_constrained::{closure#0}>>::{closure#0}, core[ad2aefeb2646d5e0]::iter::traits::iterator::Iterator::for_each::call<(rustc_hir_analysis[38d60f9cb4d84cc8]::constrained_generic_params::Parameter, ()), <hashbrown[26bc3f9f7bedacb6]::map::HashMap<rustc_hir_analysis[38d60f9cb4d84cc8]::constrained_generic_params::Parameter, (), core[ad2aefeb2646d5e0]::hash::BuildHasherDefault<rustc_hash[a52e4748ddd034eb]::FxHasher>> as core[ad2aefeb2646d5e0]::iter::traits::collect::Extend<(rustc_hir_analysis[38d60f9cb4d84cc8]::constrained_generic_params::Parameter, ())>>::extend<core[ad2aefeb2646d5e0]::iter::adapters::map::Map<core[ad2aefeb2646d5e0]::iter::adapters::flatten::FlatMap<core[ad2aefeb2646d5e0]::slice::iter::Iter<rustc_span[93bd9c9d5a358720]::def_id::DefId>, alloc[59fef786b468a15e]::vec::Vec<rustc_hir_analysis[38d60f9cb4d84cc8]::constrained_generic_params::Parameter>, rustc_hir_analysis[38d60f9cb4d84cc8]::impl_wf_check::enforce_impl_params_are_constrained::{closure#0}>, <hashbrown[26bc3f9f7bedacb6]::set::HashSet<rustc_hir_analysis[38d60f9cb4d84cc8]::constrained_generic_params::Parameter, core[ad2aefeb2646d5e0]::hash::BuildHasherDefault<rustc_hash[a52e4748ddd034eb]::FxHasher>> as core[ad2aefeb2646d5e0]::iter::traits::collect::Extend<rustc_hir_analysis[38d60f9cb4d84cc8]::constrained_generic_params::Parameter>>::extend<core[ad2aefeb2646d5e0]::iter::adapters::flatten::FlatMap<core[ad2aefeb2646d5e0]::slice::iter::Iter<rustc_span[93bd9c9d5a358720]::def_id::DefId>, alloc[59fef786b468a15e]::vec::Vec<rustc_hir_analysis[38d60f9cb4d84cc8]::constrained_generic_params::Parameter>, rustc_hir_analysis[38d60f9cb4d84cc8]::impl_wf_check::enforce_impl_params_are_constrained::{closure#0}>>::{closure#0}>>::{closure#0}>::{closure#0}>::{closure#0}>::{closure#0}>::{closure#0}>::{closure#0}>
                               at /home/matthias/vcs/github/rust_debug_assertions/library/core/src/slice/iter/macros.rs:215:27
  83:     0x7f9a69d3f65f - <core[ad2aefeb2646d5e0]::iter::adapters::map::Map<core[ad2aefeb2646d5e0]::slice::iter::Iter<rustc_span[93bd9c9d5a358720]::def_id::DefId>, rustc_hir_analysis[38d60f9cb4d84cc8]::impl_wf_check::enforce_impl_params_are_constrained::{closure#0}> as core[ad2aefeb2646d5e0]::iter::traits::iterator::Iterator>::fold::<(), <core[ad2aefeb2646d5e0]::iter::adapters::flatten::FlattenCompat<_, _>>::iter_fold::flatten<alloc[59fef786b468a15e]::vec::Vec<rustc_hir_analysis[38d60f9cb4d84cc8]::constrained_generic_params::Parameter>, (), <core[ad2aefeb2646d5e0]::iter::adapters::flatten::FlattenCompat<_, _> as core[ad2aefeb2646d5e0]::iter::traits::iterator::Iterator>::fold::flatten<alloc[59fef786b468a15e]::vec::into_iter::IntoIter<rustc_hir_analysis[38d60f9cb4d84cc8]::constrained_generic_params::Parameter>, (), core[ad2aefeb2646d5e0]::iter::adapters::map::map_fold<rustc_hir_analysis[38d60f9cb4d84cc8]::constrained_generic_params::Parameter, (rustc_hir_analysis[38d60f9cb4d84cc8]::constrained_generic_params::Parameter, ()), (), <hashbrown[26bc3f9f7bedacb6]::set::HashSet<rustc_hir_analysis[38d60f9cb4d84cc8]::constrained_generic_params::Parameter, core[ad2aefeb2646d5e0]::hash::BuildHasherDefault<rustc_hash[a52e4748ddd034eb]::FxHasher>> as core[ad2aefeb2646d5e0]::iter::traits::collect::Extend<rustc_hir_analysis[38d60f9cb4d84cc8]::constrained_generic_params::Parameter>>::extend<core[ad2aefeb2646d5e0]::iter::adapters::flatten::FlatMap<core[ad2aefeb2646d5e0]::slice::iter::Iter<rustc_span[93bd9c9d5a358720]::def_id::DefId>, alloc[59fef786b468a15e]::vec::Vec<rustc_hir_analysis[38d60f9cb4d84cc8]::constrained_generic_params::Parameter>, rustc_hir_analysis[38d60f9cb4d84cc8]::impl_wf_check::enforce_impl_params_are_constrained::{closure#0}>>::{closure#0}, core[ad2aefeb2646d5e0]::iter::traits::iterator::Iterator::for_each::call<(rustc_hir_analysis[38d60f9cb4d84cc8]::constrained_generic_params::Parameter, ()), <hashbrown[26bc3f9f7bedacb6]::map::HashMap<rustc_hir_analysis[38d60f9cb4d84cc8]::constrained_generic_params::Parameter, (), core[ad2aefeb2646d5e0]::hash::BuildHasherDefault<rustc_hash[a52e4748ddd034eb]::FxHasher>> as core[ad2aefeb2646d5e0]::iter::traits::collect::Extend<(rustc_hir_analysis[38d60f9cb4d84cc8]::constrained_generic_params::Parameter, ())>>::extend<core[ad2aefeb2646d5e0]::iter::adapters::map::Map<core[ad2aefeb2646d5e0]::iter::adapters::flatten::FlatMap<core[ad2aefeb2646d5e0]::slice::iter::Iter<rustc_span[93bd9c9d5a358720]::def_id::DefId>, alloc[59fef786b468a15e]::vec::Vec<rustc_hir_analysis[38d60f9cb4d84cc8]::constrained_generic_params::Parameter>, rustc_hir_analysis[38d60f9cb4d84cc8]::impl_wf_check::enforce_impl_params_are_constrained::{closure#0}>, <hashbrown[26bc3f9f7bedacb6]::set::HashSet<rustc_hir_analysis[38d60f9cb4d84cc8]::constrained_generic_params::Parameter, core[ad2aefeb2646d5e0]::hash::BuildHasherDefault<rustc_hash[a52e4748ddd034eb]::FxHasher>> as core[ad2aefeb2646d5e0]::iter::traits::collect::Extend<rustc_hir_analysis[38d60f9cb4d84cc8]::constrained_generic_params::Parameter>>::extend<core[ad2aefeb2646d5e0]::iter::adapters::flatten::FlatMap<core[ad2aefeb2646d5e0]::slice::iter::Iter<rustc_span[93bd9c9d5a358720]::def_id::DefId>, alloc[59fef786b468a15e]::vec::Vec<rustc_hir_analysis[38d60f9cb4d84cc8]::constrained_generic_params::Parameter>, rustc_hir_analysis[38d60f9cb4d84cc8]::impl_wf_check::enforce_impl_params_are_constrained::{closure#0}>>::{closure#0}>>::{closure#0}>::{closure#0}>::{closure#0}>::{closure#0}>::{closure#0}>
                               at /home/matthias/vcs/github/rust_debug_assertions/library/core/src/iter/adapters/map.rs:124:9
  84:     0x7f9a69d3f65f - <core[ad2aefeb2646d5e0]::iter::adapters::fuse::Fuse<core[ad2aefeb2646d5e0]::iter::adapters::map::Map<core[ad2aefeb2646d5e0]::slice::iter::Iter<rustc_span[93bd9c9d5a358720]::def_id::DefId>, rustc_hir_analysis[38d60f9cb4d84cc8]::impl_wf_check::enforce_impl_params_are_constrained::{closure#0}>> as core[ad2aefeb2646d5e0]::iter::traits::iterator::Iterator>::fold::<(), <core[ad2aefeb2646d5e0]::iter::adapters::flatten::FlattenCompat<_, _>>::iter_fold::flatten<alloc[59fef786b468a15e]::vec::Vec<rustc_hir_analysis[38d60f9cb4d84cc8]::constrained_generic_params::Parameter>, (), <core[ad2aefeb2646d5e0]::iter::adapters::flatten::FlattenCompat<_, _> as core[ad2aefeb2646d5e0]::iter::traits::iterator::Iterator>::fold::flatten<alloc[59fef786b468a15e]::vec::into_iter::IntoIter<rustc_hir_analysis[38d60f9cb4d84cc8]::constrained_generic_params::Parameter>, (), core[ad2aefeb2646d5e0]::iter::adapters::map::map_fold<rustc_hir_analysis[38d60f9cb4d84cc8]::constrained_generic_params::Parameter, (rustc_hir_analysis[38d60f9cb4d84cc8]::constrained_generic_params::Parameter, ()), (), <hashbrown[26bc3f9f7bedacb6]::set::HashSet<rustc_hir_analysis[38d60f9cb4d84cc8]::constrained_generic_params::Parameter, core[ad2aefeb2646d5e0]::hash::BuildHasherDefault<rustc_hash[a52e4748ddd034eb]::FxHasher>> as core[ad2aefeb2646d5e0]::iter::traits::collect::Extend<rustc_hir_analysis[38d60f9cb4d84cc8]::constrained_generic_params::Parameter>>::extend<core[ad2aefeb2646d5e0]::iter::adapters::flatten::FlatMap<core[ad2aefeb2646d5e0]::slice::iter::Iter<rustc_span[93bd9c9d5a358720]::def_id::DefId>, alloc[59fef786b468a15e]::vec::Vec<rustc_hir_analysis[38d60f9cb4d84cc8]::constrained_generic_params::Parameter>, rustc_hir_analysis[38d60f9cb4d84cc8]::impl_wf_check::enforce_impl_params_are_constrained::{closure#0}>>::{closure#0}, core[ad2aefeb2646d5e0]::iter::traits::iterator::Iterator::for_each::call<(rustc_hir_analysis[38d60f9cb4d84cc8]::constrained_generic_params::Parameter, ()), <hashbrown[26bc3f9f7bedacb6]::map::HashMap<rustc_hir_analysis[38d60f9cb4d84cc8]::constrained_generic_params::Parameter, (), core[ad2aefeb2646d5e0]::hash::BuildHasherDefault<rustc_hash[a52e4748ddd034eb]::FxHasher>> as core[ad2aefeb2646d5e0]::iter::traits::collect::Extend<(rustc_hir_analysis[38d60f9cb4d84cc8]::constrained_generic_params::Parameter, ())>>::extend<core[ad2aefeb2646d5e0]::iter::adapters::map::Map<core[ad2aefeb2646d5e0]::iter::adapters::flatten::FlatMap<core[ad2aefeb2646d5e0]::slice::iter::Iter<rustc_span[93bd9c9d5a358720]::def_id::DefId>, alloc[59fef786b468a15e]::vec::Vec<rustc_hir_analysis[38d60f9cb4d84cc8]::constrained_generic_params::Parameter>, rustc_hir_analysis[38d60f9cb4d84cc8]::impl_wf_check::enforce_impl_params_are_constrained::{closure#0}>, <hashbrown[26bc3f9f7bedacb6]::set::HashSet<rustc_hir_analysis[38d60f9cb4d84cc8]::constrained_generic_params::Parameter, core[ad2aefeb2646d5e0]::hash::BuildHasherDefault<rustc_hash[a52e4748ddd034eb]::FxHasher>> as core[ad2aefeb2646d5e0]::iter::traits::collect::Extend<rustc_hir_analysis[38d60f9cb4d84cc8]::constrained_generic_params::Parameter>>::extend<core[ad2aefeb2646d5e0]::iter::adapters::flatten::FlatMap<core[ad2aefeb2646d5e0]::slice::iter::Iter<rustc_span[93bd9c9d5a358720]::def_id::DefId>, alloc[59fef786b468a15e]::vec::Vec<rustc_hir_analysis[38d60f9cb4d84cc8]::constrained_generic_params::Parameter>, rustc_hir_analysis[38d60f9cb4d84cc8]::impl_wf_check::enforce_impl_params_are_constrained::{closure#0}>>::{closure#0}>>::{closure#0}>::{closure#0}>::{closure#0}>::{closure#0}>::{closure#0}>
                               at /home/matthias/vcs/github/rust_debug_assertions/library/core/src/iter/adapters/fuse.rs:91:19
  <snip>

error: the compiler unexpectedly panicked. this is a bug.

note: we would appreciate a bug report: https://github.com/rust-lang/rust/issues/new?labels=C-bug%2C+I-ICE%2C+T-compiler&template=ice.md

note: rustc 1.72.0-dev running on x86_64-unknown-linux-gnu

note: compiler flags: -Z lower-impl-trait-in-trait-to-assoc-ty

query stack during panic:
#0 [collect_return_position_impl_trait_in_trait_tys] comparing an impl and trait method signature, inferring any hidden `impl Trait` types in the process
#1 [type_of] computing type of `<impl at /home/matthias/vcs/github/rust/tests/ui/impl-trait/icemaker/5BA695890944A2D1E5C0D962DDA4006BABC66F27CECCC0A6A1719F8BAFC9F84C.rs:8:1: 8:16>::{opaque#0}`
#2 [check_mod_impl_wf] checking that impls are well-formed in top-level module
#3 [analysis] running analysis passes on this crate
end of query stack
error: aborting due to previous error; 1 warning emitted

For more information about this error, try `rustc --explain E0412`.
```

</p>
</details>

--- a/compiler/rustc_hir_analysis/src/check/compare_impl_item.rs
+++ b/compiler/rustc_hir_analysis/src/check/compare_impl_item.rs
@@ -669,11 +669,13 @@ pub(super) fn collect_return_position_impl_trait_in_trait_tys<'tcx>(
         )
         .fold_with(&mut collector);
 
-    debug_assert_ne!(
-        collector.types.len(),
-        0,
-        "expect >1 RPITITs in call to `collect_return_position_impl_trait_in_trait_tys`"
-    );
+    if !unnormalized_trait_sig.output().references_error() {
+        debug_assert_ne!(
+            collector.types.len(),
+            0,
+            "expect >1 RPITITs in call to `collect_return_position_impl_trait_in_trait_tys`"
+        );
+    }
 
     let trait_sig = ocx.normalize(&norm_cause, param_env, unnormalized_trait_sig);
     trait_sig.error_reported()?;

--- a/tests/ui/async-await/in-trait/return-not-existing-type-wrapping-rpitit.current.stderr
+++ b/tests/ui/async-await/in-trait/return-not-existing-type-wrapping-rpitit.current.stderr
@@ -1,0 +1,9 @@
+error[E0412]: cannot find type `Missing` in this scope
+  --> $DIR/return-not-existing-type-wrapping-rpitit.rs:10:25
+   |
+LL |     fn bar() -> Wrapper<Missing<impl Sized>>;
+   |                         ^^^^^^^ not found in this scope
+
+error: aborting due to previous error
+
+For more information about this error, try `rustc --explain E0412`.

--- a/tests/ui/async-await/in-trait/return-not-existing-type-wrapping-rpitit.next.stderr
+++ b/tests/ui/async-await/in-trait/return-not-existing-type-wrapping-rpitit.next.stderr
@@ -1,0 +1,9 @@
+error[E0412]: cannot find type `Missing` in this scope
+  --> $DIR/return-not-existing-type-wrapping-rpitit.rs:10:25
+   |
+LL |     fn bar() -> Wrapper<Missing<impl Sized>>;
+   |                         ^^^^^^^ not found in this scope
+
+error: aborting due to previous error
+
+For more information about this error, try `rustc --explain E0412`.

--- a/tests/ui/async-await/in-trait/return-not-existing-type-wrapping-rpitit.rs
+++ b/tests/ui/async-await/in-trait/return-not-existing-type-wrapping-rpitit.rs
@@ -1,0 +1,20 @@
+// edition:2021
+// [next] compile-flags: -Zlower-impl-trait-in-trait-to-assoc-ty
+// revisions: current next
+
+#![feature(return_position_impl_trait_in_trait)]
+
+struct Wrapper<T>(T);
+
+trait Foo {
+    fn bar() -> Wrapper<Missing<impl Sized>>;
+    //~^ ERROR: cannot find type `Missing` in this scope [E0412]
+}
+
+impl Foo for () {
+    fn bar() -> Wrapper<i32> {
+        Wrapper(0)
+    }
+}
+
+fn main() {}

